auth/auth/auth.py

```diff
@@ -431,6 +545,35 @@ async def delete_user(request, userdata):  # pylint: disable=unused-argument
     return web.HTTPFound(deploy_config.external_url('auth', '/users'))
 
 
+@routes.delete('/api/v1alpha/user/{user}')
+@rest_authenticated_developers_only
+async def delete_user(request: web.Request, userdata):  # pylint: disable=unused-argument
+    db = request.app['db']
+    username = request.match_info['user']
+
+    n_rows = await db.execute_update(
+        '''
+UPDATE users
+SET state = 'deleting'
+WHERE username = %s;
+''',
+        username,
+    )
+    if n_rows != 1:
+        assert n_rows == 0
+        raise web.HTTPBadRequest(body=f"Delete failed, no such user '{username}'")
+
+    delay = 5
+    while True:
+        user = await user_from_username(db, username)
+        if user['state'] != 'deleting':
+            break
+        delay = await sleep_and_backoff(delay, max_delay=30)
```

Comment:
No polling, the request should return immediately. If the user wants to poll for deletion, they can do that separately.